tendrl/gluster_integration/objects/volume/atoms/generate_brick_mapping/__init__.py

```diff
@@ -0,0 +1,114 @@
+import json
+
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons import objects
+from tendrl.commons.objects.job import Job
+
+
+class GenerateBrickMapping(objects.BaseAtom):
+    def __init__(self, *args, **kwargs):
+        super(GenerateBrickMapping, self).__init__(*args, **kwargs)
+
+    def run(self):
+        Event(
+            Message(
+                priority="info",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "Generating brick mapping for gluster volume"
+                },
+                job_id=self.parameters["job_id"],
+                flow_id=self.parameters["flow_id"],
+                cluster_id=NS.tendrl_context.integration_id,
+            )
+        )
+        brick_count = self.parameters.get('Volume.brick_count')
+        subvol_size = self.parameters.get('Volume.subvol_size')
+        message = ""
+        # get brick_count number of bricks from all the selected nodes
+
+        nodes = {}
+        for node in self.parameters.get('Cluster.node_configuration'):
+            key = "nodes/%s/NodeContext/fqdn" % node
+            host = NS._int.client.read(key).value
+            nodes[host] = []
+        
+        bricks = NS._int.client.read(
+            '/clusters/%s/Bricks/free/' % NS.tendrl_context.integration_id
+        )
+        for brick in bricks.leaves:
+            brick = brick.key.split("/")[-1]
+            if brick.split(":")[0] in nodes:
+                if len(nodes[brick.split(":")[0]]) < brick_count:
+                    nodes[brick.split(":")[0]].append(brick)
+
+        # Form a brick list such that when you fill sub volumes with
+        # bricks from this list, it should honour the failure domains
+
+        brick_list = []
+        total_bricks = len(nodes) * brick_count
+        for iterator in range(total_bricks):
+            brick_list.append("")
+        
+        counter = 0
+        node_count = len(nodes)
+        for key,value in nodes.iteritems():
+            if len(value) < brick_count:
+                message = "Host %s has %s bricks which is less than bricks per host %s" % (
+                    key, len(value), brick_count
+                )
+                job = Job(job_id=self.parameters["job_id"]).load()
+                res = {"message": message,"result": [[]], "optimal": False}
+                job.output["GenerateBrickMapping"] = json.dumps(res)
```

Comment:
If control enters this block it means the user provided parameters are faulty and we go out of the function. So the control will enter this block only once.
Not fully clear if that's what you are concerned about